src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -607,16 +608,28 @@ static boolean requiresChangeParentUpdateAction(@Nonnull final Category category
         }
     }
 
+    private CompletionStage<Void> fetchAndUpdate(@Nonnull final Category oldCategory,
+                                                 @Nonnull final CategoryDraft newCategory) {
+        final String key = oldCategory.getKey();
```

Comment:
`getKey()` is nullable, but `fetchCategory()` category requires non-null. Either verify before supply, or make `fetchCategory()` accept nullable (and directly return empty optional in this case).
IMHO the last approach is more flexible.

Reply:
I was thinking about this actually. However, this method is a private helper method and is used in execution where key won't be `null`. So I wasn't sure:
1. Why check for `null`, when we are sure at this point that it is not `null` because its checked earlier on.. Or
2. Check anyways, just in case if the method is used for some other implementation in the future.
In general, What is the better approach in such cases?

Reply:
ok, i understand the approach with "we are sure at this point that it is not null". 

In the same it i'd _recommend_ (but not require) in general to implement the service more flexible, e.g. allow to accept **null** and return empty optional. For example, in most of our services in [`commercetools-paypal-plus-integration`](https://github.com/commercetools/commercetools-paypal-plus-integration/tree/master/src/main/java/com/commercetools/service/ctp/impl) project we accept `@Nullable`, but just return empty values in this case, so the future service consumers don't care about arguments verification.